PythonVisualizations/LinkedList.py

```diff
@@ -1,451 +1,585 @@
+                self.wait(1.0)
+
+                # update the display
+                self.cleanUp(callEnviron)
+                return pos
+
+            # if the value hasn't been found, wait and then move the arrow over one cell
+            self.wait(0.7)
+            cur = cur.next
+           
+            arrow = (arrow, )
+            if (pos-4)%self.LEN_ROW == 0: 
+                self.moveItemsBy(arrow, (-((self.CELL_WIDTH + self.CELL_GAP)*(self.LEN_ROW-1)), (self.CELL_HEIGHT + self.ROW_GAP)))
+            else: 
+                self.moveItemsBy(arrow, (self.CELL_WIDTH + self.CELL_GAP,0))
+            pos+=1
+            
+        self.cleanUp(callEnviron)
+        return None
+
+    def display_neatly(self):
+        self.canvas.delete("all")
+        self.firstPointer()
+        cur = self.first
+        pos = 1
+        while cur:
+            val = cur.key
+            self.createCell(pos, '12', val, cur)
+            cur = cur.next
+            pos += 1
+        self.window.update()
+              
+    
+    def cleanUp(self, *args, **kwargs): 
+        super().cleanUp(*args, **kwargs)
+
+            
+    
+    ### BUTTON FUNCTIONS##
+    def clickFind(self):
+        val = self.validateArgument()
+        if val is None:
+            self.setMessage("Input value must be less than 9 characters")
+        else:
+            result = self.find(val)
+            if result != None:
+                msg = "Found {}!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+            self.setMessage(msg)
+            
+        self.clearArgument()
+    
+    def clickInsert(self):
+        val = self.validateArgument()
+        if val is None:
+            self.setMessage("Input value must be less than 9 characters")
+        elif len(self) >= self.MAX_SIZE:
+            self.setMessage("Error! Linked List is already full.")
+        else:  
+            self.insertElem(val)
+
+
+
+    def clickDelete(self):
+        val = self.validateArgument()
+        msg = None
+        if not self.first:
+            msg = "ERROR: Linked list is empty"
+        elif not val:
+            msg = "Input value must be less than 9 characters" 
+        elif val == -1:
+            result = self.delete()
+            msg = "first node deleted"
+        else:
+            result = self.delete(val)
+            if result != None:
+                msg = "{} Deleted!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+        self.setMessage(msg)
+        self.clearArgument()
+        
+    def clickNewLinkedList(self):
+        self.canvas.delete('all')
+        self.newLinkedList()
+    
+    def clickGetFirst(self):
+        cur = self.getFirst()
+        if cur!= None: 
+            val = cur.key
+            msg = "The first node is {}".format(val)
+        else: msg = "ERROR: Linked list is empty"
+        self.setMessage(msg)
+        self.clearArgument()
```

Comment:
Please add an animation for Get First that behaves like the Peek operation for Stacks (copying the item value to an output area).